java/java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java

```diff
@@ -4880,6 +4886,14 @@ private void addClassModifiers(Env env, Set<Modifier> modifiers) {
         }
     }
 
+    private static boolean contains(Set<Modifier> modifiers, String modifier) {
+        try {
+            return modifiers.contains(Modifier.valueOf(modifier));
+        } catch (IllegalArgumentException ex) {
```

Comment:
How performance critical is this? I read somewhere, that exception handling is not the fast path in the JVM, so if this is a problem we'd need to buffer that.

Reply:
it shouldn't be a problem there since its called only a handful of times per completion. But i do agree in principle, exceptions should be avoided if possible. 

Additionally: it won't throw with recent javacs, be it auto-javac or JDK's. If java 17 javac is standard we can remove all the workarounds and use the enum directly.

Reply:
Thanks for confirming.